cli/whoami.ts

```diff
@@ -0,0 +1,37 @@
+import type {ArgumentsCamelCase} from 'yargs';
+import {createCodaClient} from './helpers';
+import {formatEndpoint} from './helpers';
+import {getApiKey} from './config_storage';
+import {isResponseError} from '../helpers/external-api/coda';
+import {printAndExit} from '../testing/helpers';
+import {tryParseSystemError} from './errors';
+
+interface WhoamiArgs {
+  apiToken?: string;
+  codaApiEndpoint: string;
+}
+
+export async function handleWhoami({apiToken, codaApiEndpoint}: ArgumentsCamelCase<WhoamiArgs>) {
+  const formattedEndpoint = formatEndpoint(codaApiEndpoint);
+  if (!apiToken) {
+    apiToken = getApiKey(codaApiEndpoint);
+    if (!apiToken) {
+      return printAndExit('Missing API key. Please run `coda register` to register one.');
+    }
+  }
+
+  const client = createCodaClient(apiToken, formattedEndpoint);
+
+  try {
+    const {name, loginId, workspace} = await client.whoami();
+
+    printAndExit(`You are ${name} (${loginId}) in workspace ${workspace.id}`, 0);
```

Comment:
Is it returning a workspace for you in practice? I can't remember exactly but I thought maybe the token would only include a workspace if it's scoped to a doc. I don't think a workspace makes any sense if it's a general API token or scoped to a pack.

We should also print info about the scope: is it read-only vs read-write. Is it scoped to all packs or just a specific pack id. Is it scoped to something doc-related, in which case indicate that the token cannot be used with the packs CLI.

And nit, `return`.

Reply:
`whoami` always returns an workspace but doesn't return information about what a token is scoped to or whether it's read-only vs read-write:

https://github.com/coda/coda/blob/b446d841c21cf3d078fc2cb4da7051af2d9c9b5f/modules/server/api/private/actions/v1/whoami.ts#L49

Can I determine any of that scoping information by parsing the token itself, and if so would it be appropriate to parse the token in client-side SDK code?

Reply:
we don't seem to have an API doing that. the packs-sdk itself doesn't have the creds to decode the API token.